echopype/tests/clean/test_noise.py

```diff
@@ -2,11 +2,433 @@
+        "the overhead sorting."
+    )
+    assert any(expected_warning in record.message for record in caplog.records)
+
+    # Turn off logger verbosity
+    ep.utils.log.verbose(override=True)
+
+    # Check for func value error:
+    with pytest.raises(ValueError, match="Input `func` is `nanmode`. `func` must be `nanmean` or `nanmedian`."):
+        ep.clean.mask_transient_noise(
+            ds_Sv,
+            depth_bin=depth_bin,
+            num_side_pings=num_side_pings,
+            exclude_above=exclude_above,
+            func="nanmode",
+        )
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    ("chunk", "func"),
+    [
+        (False, "nanmean"),
+        (True, "nanmean"),
+        (False, "nanmedian"),
+        (True, "nanmedian"),
+    ],
+)
+def test_calc_transient_noise_pooled_Sv_values(chunk, func):
+    """
+    Manually check if the pooled Sv for transient noise masking contains 
+    the correct nan boundary and the correct bin aggregate values.
+    """
+    # Open raw, calibrate, and add depth
+    ed = ep.open_raw(
+        "echopype/test_data/ek60/from_echopy/JR230-D20091215-T121917.raw",
+        sonar_model="EK60"
+    )
+    ds_Sv = ep.calibrate.compute_Sv(ed)
+    ds_Sv = ep.consolidate.add_depth(ds_Sv)
+
+    # Select Sv subset
+    ds_Sv = ds_Sv.isel(ping_time=slice(294, 300), range_sample=slice(1794,1800))
+
+    if chunk:
+        # Chunk calibrated Sv
+        ds_Sv = ds_Sv.chunk("auto")
+
+    # Set window args
+    depth_bin = 0.2 # depth values ~0.2m apart per range sample
+    num_side_pings=2
+    exclude_above = 250
+
+    # Compute pooled Sv
+    pooled_Sv = calc_transient_noise_pooled_Sv(
+        ds_Sv, func, depth_bin, num_side_pings, exclude_above
+    ).compute()
+
+    # Compute min and max values
+    depth_values_min = ds_Sv["depth"].min()
+    depth_values_max = ds_Sv["depth"].max()
+    ping_time_index_min = 0
+    ping_time_index_max = len(ds_Sv["ping_time"])
+
+    # Create ping time indices array
+    ping_time_indices = xr.DataArray(
+        np.arange(len(ds_Sv["ping_time"]), dtype=int),
+        dims=["ping_time"],
+        coords=[ds_Sv["ping_time"]],
+        name="ping_time_indices",
+    )
+
+    # Check appropriate NaN boundaries
+    within_mask = (
+        (ds_Sv["depth"] - depth_bin >= depth_values_min) &
+        (ds_Sv["depth"] + depth_bin <= depth_values_max) &
+        (ds_Sv["depth"] - depth_bin >= exclude_above) &
+        (ping_time_indices - num_side_pings >= ping_time_index_min) &
+        (ping_time_indices + num_side_pings <= ping_time_index_max)
+    )
+    unique_pool_boundary_values = np.unique(
+        pooled_Sv.where(~within_mask)
+    )
+
+    # Check that NaN is the only pool boundary unique value
+    assert np.isclose(unique_pool_boundary_values, np.array([np.nan]), equal_nan=True)
+
+    # Compute Sv
+    ds_Sv["Sv"] = ds_Sv["Sv"].compute()
+
+    # Manually check correct binning and aggregation values
```

Comment:
For the checks here: are you doing spot-check only because the test would run for too long? In the above you actually calculated `pooled_Sv`, so I am just curious why you didn't end up check all values.

Reply:
Oh, I am doing spot-checking because the code I wrote for computing `pooled_Sv` was the simplest implementation I could think of for computing it programmatically at the time. So my thought process then was to just manually check a few values. But now that I look at it, the array is small enough that I can manually check all values, which I'll add in this revision.